cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -633,7 +633,19 @@ public function attachment_url( $url, $attachment_id ) {
 		if ( false !== $previous_url ) {
 			return substr( $url, $previous_url );
 		}
-		if ( ! doing_action( 'wp_insert_post_data' ) && false === $this->in_downsize && ! apply_filters( 'cloudinary_doing_upload', '__return_false' ) ) {
+		if (
+			! doing_action( 'wp_insert_post_data' )
+			&& false === $this->in_downsize
+			/**
+			 * Filter doing upload.
+			 * If so, return the default attachment URL.
+			 *
+			 * @param bool Default false.
+			 *
+			 * @return bool
+			 */
+			&& ! apply_filters( 'cloudinary_doing_upload', false )
```

Comment:
One small correction here:

```suggestion
			&& apply_filters( 'cloudinary_doing_upload', false )
```

Otherwise we default to true. Thanks for addressing this 😁

Reply:
@dugajean 

The default should be true. We should almost always resolve to serve the Cloudinary. However, in the case of uploading, we should bail any change.